webhook/webhook.go

```diff
@@ -0,0 +1,59 @@
+package webhook
+
+import (
+  "crypto/hmac"
+  "crypto/sha256"
+  "encoding/hex"
+  "time"
+)
+
+const (
+  DefaultTimeoutSeconds = 10
+  MinHttpStatusErrorCode = 400
+)
+
+type Webhook struct {
+  client *http.Client
+}
+
+//TODO
+
+
+func (wh *Webhook) Send(data interface{}) error {
+  jsonData, err := json.Marshal(data)
+  if err != nil {
+    log.Error(err)
+    return err
+  }
+  req, err := http.NewRequest("POST", wh.Url, bytes.NewBuffer(jsonData))
+  ts := int32(time.Now().Unix())
+  signat := wh.sign(data, ts)
+  req.Header.Set("X-Gophish-Signature", signat)
+  req.Header.Set("Content-Type", "application/json")
+  resp, err := wh.client.Do(req)
+  if err != nil {
+    log.Error(err)
+    return err
+  }
+  defer resp.Body.Close()
+  if resp.Status >= MinHttpStatusErrorCode {
```

Comment:
TIL about `MinHttpStatusErrorCode`. Nice!

Reply:
Oh, nvm, I see you defined it. I thought it was part of the standard `net/http` package 😝